client/server/pages/index.js

```diff
@@ -296,6 +297,17 @@ function setUpLoggedOutRoute( req, res, next ) {
 			subscriptionManagementSubkey: req.cookies.subkey,
 		};
 	}
+	const acceptLang = req.get( 'Accept-Language' )?.split( ',' )[ 0 ];
```

Comment:
I noticed that as a side-effect of this change, some other routes are now being localized based on the detected language from the `Accept-Language` header. For example, `/accept-invite` and `/pricing` from Jetpack Cloud are now localized, and some other routes like `/devdocs` get partially localized - e.g. the `lang` and `dir` attributes on the `html` tag are set based on the browser's language, but the rest of the page is rendered in English.

Is setting the locale on the server side necessary? If not, I suppose the locale can then just be set on the client side, similarly to how we do it on `/start/user` with the [LocaleSuggestions component](https://github.com/Automattic/wp-calypso/blob/trunk/client/components/locale-suggestions/index.jsx#L31).